code/modules/storage/storage.dm

```diff
@@ -0,0 +1,413 @@
+/// hide storage
+/datum/storage/proc/hide_hud(mob/user)
+	if (user.s_active == src.hud)
+		user.s_active = null
+		user.detach_hud(src.hud)
+
+// ----------------- PUBLIC PROCS ----------------------
+
+/// check if the storage can hold an item or not
+/datum/storage/proc/check_can_hold(obj/item/W)
+	if (!W)
+		return STORAGE_CANT_HOLD
+
+	if (W.cant_drop)
+		return STORAGE_WONT_FIT
+
+	// if can_hold is defined, check against that
+	if (length(src.can_hold))
+		// early skip if weight class is allowed
+		if (src.check_wclass && W.w_class <= src.max_wclass)
+			return STORAGE_CAN_HOLD
+		for (var/type in src.can_hold)
+			if (ispath(type) && istype(W, type))
+				return STORAGE_CAN_HOLD
+		for (var/type in src.can_hold_exact)
+			if (ispath(type) && W.type == type)
+				return STORAGE_CAN_HOLD
+		return STORAGE_CANT_HOLD
+
+	else if (W.w_class > src.max_wclass)
+		return STORAGE_WONT_FIT
+
+	if (src.is_full())
+		return STORAGE_IS_FULL
+
+	return STORAGE_CAN_HOLD
+
+/// when adding an item in
+/datum/storage/proc/add_contents(obj/item/I, mob/user = null, visible = TRUE)
+	if (user?.equipped() == I)
+		user.u_equip(I)
+	src.stored_items += I
+	I.set_loc(src.linked_item)
+	src.hud.add_item(I, user)
+	I.stored = src
+
+	src.add_contents_extra(I, user, visible)
+
+/// available if add_contents needs to be overridden
+/datum/storage/proc/add_contents_extra(obj/item/I, mob/user, visible)
+	// make sure storage item tooltip will be updated
+	if (istype(src.linked_item, /obj/item))
+		var/obj/item/W = src.linked_item
+		W.tooltip_rebuild = TRUE
+	// for storages that change icon with contents
+	src.linked_item.UpdateIcon()
+	// a mob put the item in
+	if (!istype(user))
+		return
+	src.linked_item.add_fingerprint(user)
+	if (visible)
+		animate_storage_rustle(src.linked_item)
+		if (!src.sneaky && !istype(I, /obj/item/gun/energy/crossbow))
+			user.visible_message("<span class='notice'>[user] has added [I] to [src.linked_item]!</span>",
+				"<span class='notice'>You have added [I] to [src.linked_item].</span>")
+		playsound(src.linked_item.loc, "rustle", 50, TRUE, -5)
+
+/// when transfering something in the storage out
+/datum/storage/proc/transfer_stored_item(obj/item/I, atom/location, add_to_storage = FALSE, mob/user = null)
+	if (!(I in src.get_contents()))
+		return
+	src.stored_items -= I
+	src.hud.remove_item(I, user)
+	I.stored = null
+
+	src.transfer_stored_item_extra(I, location, add_to_storage, user)
+
+/// for use if transfer_stored_item is overridden
+/datum/storage/proc/transfer_stored_item_extra(obj/item/I, atom/location, add_to_storage, mob/user)
+	// update storage item tooltip
+	if (istype(src.linked_item, /obj/item))
+		var/obj/item/W = src.linked_item
+		W.tooltip_rebuild = TRUE
+	src.linked_item.UpdateIcon()
+	if (location?.storage && add_to_storage)
+		location.storage.add_contents(I, user)
+	else
+		I.set_loc(location)
+		if (isturf(location))
+			I.dropped()
+
+/// return outputtable capacity
+/datum/storage/proc/get_capacity()
```

Comment:
would prefer if this was named `get_capacity_string` or something, since `get_capacity` sounds like it's going to return an integer. Should be an easy find-replace

Reply:
https://github.com/goonstation/goonstation/pull/13023/commits/8213d00ba851caaf48ded5bbcdfa2c441b20bc70